python/ray/data/dataset_pipeline.py

```diff
@@ -99,13 +101,22 @@ def iter_batches(self,
         """
 
         def gen_batches() -> Iterator[BatchType]:
+            time_start = time.perf_counter()
+
             for ds in self.iter_datasets():
+                wait_start = time.perf_counter()
                 for batch in ds.iter_batches(
                         prefetch_blocks=prefetch_blocks,
                         batch_size=batch_size,
                         batch_format=batch_format,
                         drop_last=drop_last):
-                    yield batch
+                    self._stats.iter_wait_s.add(time.perf_counter() -
+                                                wait_start)
+                    with self._stats.iter_user_s.timer():
+                        yield batch
+                    wait_start = time.perf_counter()
+
+            self._stats.iter_total_s.add(time.perf_counter() - time_start)
```

Comment:
Same question here: any reason for not using the .timer() context manager for this one? Seems like all of this could be under that context manager.
```python
with self._stats.iter_total_s.timer():
    for ds in self.iter_datasets():
        ...
```

Reply:
Same.